storage/sqlite/src/lib.rs

```diff
@@ -227,13 +227,13 @@ impl backend::BackendImpl for SqliteImpl {}
 
 #[derive(Eq, PartialEq, Clone, Debug)]
 pub struct Sqlite {
-    path: PathBuf,
+    path: Option<PathBuf>,
```

Comment:
It's much better to use a custom enum for this. Something like:

```
enum SqliteBackend {
    InMemory,
    File(PathBuf),
}
```

Reply:
Much better indeed